src/extensions/ExtensionManager.as

```diff
@@ -323,6 +329,9 @@ public class ExtensionManager {
 			if(b.isRequester) {
 				if(b.requestState == 2) {
 					b.requestState = 0;
+					// TLF: should it increase the blocking count here instead...?
+					// If so then it'll need also increasing in resetArgsBlockingCount
+					// for any blocks which are requesters.
```

Comment:
Sorry I missed this question!
There's a lot of subtlety around requesters and yielding. It's possible I'm wrong about not needing to increase blocking count here, but I still think it's true.
Specifically, I think:
* Trying to read the value when `requestState==0` should cause a request and a yield, and `requestState` should go to 1.
* Trying to read the value when `requestState==1` should just yield.
* Trying to read the value when `requestState==2` should succeed and not yield, and `requestState` should go to zero.

I think of the yield as being associated with a pending request (basically, `requestState==1`). When the state transitions from 2 to 0 there's no active request, so no need to yield. It might be that a request starts immediately afterward, as in the "`else if(b.requestState == 0) {`" case below. If that happens we should yield at that point, but that's not necessarily the case.